File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1
        if x>=0 :
            sign = 1
        temp = x*sign
        res = 0
        while(temp>0):
            res = res*10+temp%10
            temp//=10
        if res < -2**31 or res > 2**31 - 1:
            return 0
        return res*sign
    
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        temp = self.reverse(x)
        if temp==x :
            return True
        return False

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0) return false;
        int temp = reverse(x);
        if(x==temp) return true;
        return false;
    }
    public int reverse(int x) {
        long res = 0;
        int sign = x>=0 ? 1 : -1;
        int temp = x*sign;
        
        while(temp>0){
            res = res*10+temp%10;
            temp/=10;
        }
        if(res<Integer.MIN_VALUE || res>Integer.MAX_VALUE) return 0;
        else return (int) sign*(int)res;
    }
}